src/gt4py/next/program_processors/runners/gtfn.py

```diff
@@ -29,17 +28,14 @@
 
 # TODO(ricoh): Add support for the whole range of arguments that can be passed to a fencil.
 def convert_arg(arg: Any) -> Any:
+    if (origin := getattr(arg, "__gt_origin__", None)) is not None:
+        return arg.ndarray, origin
```

Comment:
```suggestion
        assert isinstance(arg, common.Field)
        return arg.ndarray, origin
```
For correctness & readability when executing in debug mode.